qiskit/algorithms/optimizers/spsa.py

```diff
@@ -190,17 +192,60 @@ def __init__(
             callback: A callback function passed information in each iteration step. The
                 information is, in this order: the number of function evaluations, the parameters,
                 the function value, the stepsize, whether the step was accepted.
+            termination_checker: A callback function executed at the end of each iteration step. The
+                arguments are, in this order: the parameters, the function value, the number
+                of function evaluations, the stepsize, whether the step was accepted. If the callback
+                returns True, the optimization is terminated.
+                To prevent additional evaluations of the objective method, if the objective has not yet
+                been evaluated, the objective is estimated by taking the mean of the objective
+                evaluations used in the estimate of the gradient.
+
 
         Raises:
             ValueError: If ``learning_rate`` or ``perturbation`` is an array with less elements
                 than the number of iterations.
+
+        Example:
+            .. jupyter-execute::
```

Comment:
If we put this into a `jupyter-execute` it gets executed each time we build the documentation, I think it'd be better to just show the code 
```suggestion
            .. code-block::python
```